fix: several disabled pylint rules in models/helpers.py

kkucharc opened the pull request:
### SUMMARY
Fixed several disabled pylint rules in models/helpers.py:
- pylint import list seems not conflicting with isort anymore
- changed escaping of `[` to double backslash
- changed private method which is used as public
- defined more specific exception handling in JSON loading
- removed unnecessary error handling and replaced with condition

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TEST PLAN
<!--- What steps should be taken to verify the changes -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

kkucharc committed:
Removed conflicting lint and isort check in model helpers seems it's not appearing anymore

kkucharc committed:
Removed disabled linting for accessing private method. `parent_foreign_key_mappings` becomes public because it is accessed by other instance than `self`.

kkucharc committed:
Updated model's helper - removed unecessary exception and replaced with check while accessing global context to reset ownerships.

kkucharc committed:
Updated model's helper - renamed unused attribute to private in user link method.

kkucharc committed:
Updated model's helper - added specific exception for adding extra json column. Removed disabled pylint rule.

kkucharc commented on superset/models/helpers.py:
```diff
@@ -66,14 +63,6 @@ class ImportMixin:
     # The names of the attributes
     # that are available for import and export
 
-    @classmethod
```
@willbarrett I am really interested in your opinion about this change. I assume this method is private because:
- it might change in future, or
- it's supposed to be used only in `ImportMixin` class.
But since this method is used on other instances (not only `cls`) I suggested making it public. WDYT?

willbarrett replied:
Yes, I think making it public is a good idea.

kkucharc commented:
@willbarrett Would you find a time to take a look?

willbarrett reviewed (approved):
LGTM after rebase.

kkucharc commented:
@willbarrett Thanks for review. 
I just rebased it.

dpgaspar commented on superset/models/helpers.py:
```diff
@@ -424,7 +425,10 @@ class ExtraJSONMixin:
-        except Exception:  # pylint: disable=broad-except
+        except (TypeError, JSONDecodeError) as exc:
+            logger.error(
+                "Unable to load an extra json: %r. Leaving empty.", exc, exc_info=True
```
Can `exc` or `exc_info` log data from `self.extra_json` ?

kkucharc replied:
Good idea! I will add it to an exception logger.

dpgaspar replied:
maybe I did not explain myself right, sorry. I want to make sure that `self.extra_json` never gets logged by `TypeError` or `JSONDecodeError` by `exc` or `exc_info`

kkucharc replied:
@dpgaspar do you mind to take a look again? I have one doubt: do you think is it possible there could be raised exception while accessing `self.extra_json` for logger?

kkucharc replied:
Och, I see, then I just did the opposite thing! :)

kkucharc replied:
Just to be sure I check that and any of them won't show what `self.extra_json` contains.
`JSONDecodeError` should something like:
`json.decoder.JSONDecodeError: Unterminated string starting at: line 1 column 11 (char 10)`
with stacktrace.
And `TypeError` something like this for example:
`TypeError: the JSON object must be str, bytes or bytearray, not dict`

dpgaspar replied:
nice then, thks

dpgaspar commented on superset/models/helpers.py:
```diff
@@ -321,11 +322,11 @@ def template_params_dict(self) -> Dict[Any, Any]:
 
-def _user_link(user: User) -> Union[Markup, str]:  # pylint: disable=no-self-use
-    if not user:
+def _user_link(_user: User) -> Union[Markup, str]:
```
Can't we just remove the `# pylint: disable=no-self-use` on this one?

kkucharc replied:
You are absolutely right! Somehow I had to mistake that for `unused-argument`. Thanks! I'm changing it right-away.

dpgaspar reviewed (commented):
left a couple of comments

kkucharc committed:
Applied changes after review to `models/helpers.py`:
- removed unecesary function's param rename
- added extra JSON content in exception

kkucharc committed:
Removed self.extra_json content from exception message.
